File: ProjetASI2/Personal-Finance-Management-Spring-boot/src/main/java/com/PersonalFinanceManagementApplication/Personal/Finance/Management/Application/Model/CategoryLimitHistory.java
package com.PersonalFinanceManagementApplication.Personal.Finance.Management.Application.Model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "Category_Limit_History")
@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class CategoryLimitHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne
    @JoinColumn(name = "person_id")
    @JsonIgnore
    private Person person;


    private BigDecimal monthlyLimit;

    private BigDecimal totalSpent;

    private LocalDateTime dateModified;

    public CategoryLimitHistory(Category category, Person person, BigDecimal monthlyLimit, BigDecimal totalSpent, LocalDateTime dateModified) {
        this.category = category;
        this.person = person;
        this.monthlyLimit = monthlyLimit;
        this.totalSpent = totalSpent;
        this.dateModified = dateModified;
    }
}

File: ProjetASI2/PersonalFinanceManagementAppFrontEnd/src/app/Model/CategoryLimitHistory.ts
import { Category } from "./Category";
import { Person } from "./Person";

export class CategoryLimitHistory{

    id:number;
    category:Category;
    person: Person | null;
    monthlyLimit:Number;
    totalSpent: Number;
    dateModified : Date | String;

    constructor(id:number, category:Category, person: Person | null, monthlyLimit:Number, totalSpent: Number, dateModified : Date| String){
        this.id = id;
        this.person = person;
        this.category = category;
        this.monthlyLimit = monthlyLimit;
        this.totalSpent = totalSpent;
        this.dateModified = dateModified;
    }

}